### PR TITLE
oauth: Use the new and improved Sigstore redirect page

### LINKED ISSUE #354: Use the new and improved Sigstore redirect page
We should crib the work done in https://github.com/sigstore/sigstore/pull/876 and reuse it here!

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
Closes #354

### CODE REVIEW

File: sigstore/_internal/oidc/oauth.py [hunk 1]
```diff
@@ -43,13 +43,64 @@
...
+        </div>
+      </div>
+    </div>
+    <script>
+      document.getElementById("favicon").setAttribute("href", "data:image/svg+xml," + encodeURIComponent(document.getElementById("logo").outerHTML));
+    </script>
+  </body>
 </html>
-"""
+"""  # noqa: E501
```
Comment (tetsuo-cpp) [inline sigstore/_internal/oidc/oauth.py:103]: I've just decided to silence Ruff here as I don't think it helps readability to insert line breaks.

**Comment by tetsuo-cpp** [comment]
Thanks @julienv3, this looks great.

**Comment by jku** [comment]
This is (mostly?) code copied from sigstore/sigstore right? Should include some authorship info in the commit as well -- it's good for tracking copyrights but also for the authors internet fame and fortune (not everyone cares about that but it's still nice).

Adding a "Co-authored-by:" to the commit message would trigger at least github noticing the contribution, maybe that's good enough?

**Comment by tetsuo-cpp** [comment]
> This is (mostly?) code copied from sigstore/sigstore right? Should include some authorship info in the commit as well -- it's good for tracking copyrights but also good for the authors internet fame and fortune (not everyone cares about that but it's still nice).
> 
> Adding a "Co-authored-by:" to the commit message would trigger at least github noticing the contribution, maybe that's good enough?

My mistake, I completely forgot about that. I'll add the "co-authored-by" to attribute the commit to @julienv3 and add a comment in the code to explain where it's from and who wrote it. How does that sound?

**Review by di — APPROVED** [review]

**Review by woodruffw — APPROVED** [review]

**Comment by woodruffw** [comment]
LGTM! Per #361 we might to eventually move this to a stand-alone template or file, since that might help with automatic syncing against `cosign`.

**Pull request merged**